sql/analyzer/validation_rules.go

```diff
@@ -246,10 +246,29 @@ func validateGroupBy(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (s
 			groupBys = append(groupBys, expr.String())
 		}
 
+		var curAliases = make(map[string]sql.Expression)
+		if p, ok := n.Child.(*plan.Project); ok {
+			for _, projection := range p.Projections {
+				if a, ok := projection.(*expression.Alias); ok {
+					curAliases[a.Name()] = a.Child
+				}
+			}
+		}
+
 		for _, expr := range n.SelectedExprs {
 			if _, ok := expr.(sql.Aggregation); !ok {
 				if !expressionReferencesOnlyGroupBys(groupBys, expr) {
-					return nil, ErrValidationGroupBy.New(expr.String())
+					depExprs := expressionReferencesAlias(expr)
+					for _, dep := range depExprs {
+						if !expressionReferencesOnlyGroupBys(groupBys, dep) {
+							if e, ok := curAliases[dep.String()]; ok {
```

Comment:
Looks like you are using the string value, just computing it again. Use the 2-value range expression instead. Or, change the `expressionReferencesAlias` method to return just a slice of sql.Expr